jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageReader.java

```diff
@@ -81,6 +81,38 @@ static void verifyImageMetadata(ImageMetadataTemplate metadata, Path metadataCac
     this.cacheStorageFiles = cacheStorageFiles;
   }
 
+  /**
+   * Returns {@code true} if all image layers described in a manifest have a corresponding file
+   * entry in the cache.
+   *
+   * @param manifest the image manifest
+   * @return a boolean
+   */
+  boolean allLayersCached(ManifestTemplate manifest) {
+
+    List<DescriptorDigest> layerDigests;
+
+    if (manifest instanceof V21ManifestTemplate) {
+      layerDigests = ((V21ManifestTemplate) manifest).getLayerDigests();
+    } else if (manifest instanceof BuildableManifestTemplate) {
+      layerDigests =
+          ((BuildableManifestTemplate) manifest)
+              .getLayers().stream()
+                  .map(BuildableManifestTemplate.ContentDescriptorTemplate::getDigest)
+                  .collect(Collectors.toList());
+    } else {
+      throw new IllegalArgumentException("Unknown manifest type: " + manifest);
+    }
```

Comment:
I wonder if this would break current behavior for those doing multi-platform image building? For context https://github.com/GoogleContainerTools/jib/pull/2730, we also do caching for manifest lists in addition to single manifests.

Reply:
That’s a good callout, and please let me know if I'm misunderstanding anything here! 

Right now the added logic in `PullBaseImageStep.getCachedBaseImages()` never passes anything of `V22ManifestList` type into `areAllLayersCached()` calls explicitly. In the case of manifest lists, this check is [made individually](https://github.com/GoogleContainerTools/jib/pull/3767/files/ac8a2c7f45bb227e9f35f373510356c524c0ba7c#diff-220c5d50240f696a44f029f651cf4ada15dc2e82baa15085668f2bb7fbaf7cdeR495-R499) when looping over the platform-specific manifests, and returns an overall cache miss if any of the platform-specific manifests has incomplete layers. 


But, do you think there is value here to have `CacheStorageReader.areAllLayersCached()` itself handle the manifest list type, rather than rely on the code calling it? Perhaps instead of throwing an exception here, it can also just return false, and leave the rest of the behavior to existing logic?

Reply:
Sounds good!